ml-conversational-analytic-tool/githubDataExtraction.py

```diff
@@ -1,154 +1,208 @@
 
 import utils
 
 
 class GithubDataExtractor:
-    def __init__(self, access_token):
+    def __init__(self, access_token, organization):
         """
         Constructor requires an access token to start a Github session, and specifies instance variables
         """
-        self.g_ses = Github(access_token)  # Github object is used as a channel to the Github API
-        self.current_repo = None  # Current Opened Repo
         self.reaction_flag = False
-        self.repo_opened = False  # Flag to store state of repo as opened (True) or closed (False)
-        self.repo_name = ""
-        self.organization = ""
+        self.organization = organization
+        self.query_repos = self.load_query('repos.graphql')
+        self.query_pull_requests_without_reactions = self.load_query('pull_requests_without_reactions.graphql')
+        self.query_pull_requests_with_reactions = self.load_query('pull_requests_with_reactions.graphql')
+        self.headers = {'Authorization': 'bearer ' + access_token}
 
-    def openRepo(self, organization, repo_name):
+    def get_all_repos(self):
         """
-        Method to  open (access) repository with given organization and repository name (reponame).
-        Parameters: username - owner of the repository, repo_name - name of repo to be opened
+         Method to get all repos of an organization
+         Parameters: None
+         Returns: A list of repository names
+         """
+        queries_and_variables = {
+            'query': self.query_repos,
+            'variables': {'owner': self.organization}
+        }
+        repository_list = self.run_queries(query=queries_and_variables)
+        repositories = [repository.get('name') for repository in repository_list]
+        return repositories
+
+    def get_all_pull_requests(self, repo, reaction_flag=False):
         """
-        self.current_repo = self.g_ses.get_repo(organization + "/" + repo_name)  # Open repo
-        self.repo_opened = True
-        self.repo_name = repo_name
-        self.organization = organization
-        print("Opened repo {}/{}".format(organization, repo_name))
+        Method to get all pull requests from a repo
+        Parameters:
+            repo - string
+            reaction_flag - boolean
+        Returns: dataframe containing rows of pull requests
+        """
+        self.reaction_flag = reaction_flag
+        if self.reaction_flag:
+            query = self.query_pull_requests_with_reactions
+        else:
+            query = self.query_pull_requests_without_reactions
+
+        queries_and_variables = {
+            'query': query,
+            'variables': {'owner': self.organization, 'repo': repo}
+        }
+        pull_request_list = self.run_queries(query=queries_and_variables)
+        pull_data = (self.get_pull_features(pr) for pr in pull_request_list)
+        pull_request_df = pd.DataFrame(pull_data)
+
+        try:
+            return pull_request_df.sort_values(by=pull_request_df.columns[0])
```

Comment:
nit: Can you add a comment mentioning the name of the field by which the values are sorted?

Reply:
Done